gulp-core-build-karma/src/KarmaTask.ts

```diff
@@ -67,6 +78,30 @@ export class KarmaTask extends GulpTask<IKarmaTaskConfig> {
 
       completeCallback();
     } else {
+      // Normalize the match expression if one was specified
+      const { testMatch } = this.taskConfig; // tslint:disable-line:typedef
+      if (testMatch) {
+        let normalizedMatch: RegExp;
+        if (typeof testMatch === 'string') {
+          normalizedMatch = new RegExp(testMatch as string);
+        } else if (testMatch instanceof RegExp) {
+          normalizedMatch = testMatch;
+        } else {
+          completeCallback('The testMatch regular expression is invalid');
+          return;
+        }
+
+        // tslint:disable:max-line-length
+        const testsJsFileContents: string = [
+          `var context = require.context('${path.posix.join('..', this.buildConfig.libFolder)}', true, ${normalizedMatch.toString()});`,
+          `context.keys().forEach(context);`,
```

Comment:
I will need to test, but apparently using the --match flag, we can run a test or tests, and that is using the current tests.js approach. If it doesnt work we should file another bug. Biggest issue is that when we change to the approach of bundling each test file individually we lose performance, and it (and this is arguably a good thing) exposes more circular dependency issues. I didn't have time to fix/re-architect some internal packages to remove these circular deps so instead went with an alternative approach.